File: slideholder/src/main/java/zhan/library/slide/SlideAnimatorUpdateListener.java
package zhan.library.slide;

import android.animation.ValueAnimator;

public abstract class SlideAnimatorUpdateListener implements ValueAnimator.AnimatorUpdateListener {

    @Override
    public void onAnimationUpdate(ValueAnimator animation) {

    }

    public abstract void onSlideAnimationUpdate(ValueAnimator animation, int currentStatus);
}

File: slideholder/src/main/java/zhan/library/slide/SlideAnimatorUpdateListener.kt
package zhan.library.slide

import android.animation.ValueAnimator

abstract class SlideAnimatorUpdateListener : ValueAnimator.AnimatorUpdateListener {

    override fun onAnimationUpdate(animation: ValueAnimator) {

    }

    abstract fun onSlideAnimationUpdate(animation: ValueAnimator, currentStatus: Int)
}
